### PR TITLE
Selection menu

### LINKED ISSUE #9: build the selection menu for actions of the bot during games

### LINKED ISSUE #31: build a commands menu for actions that be requested
This is similar to issue #9 

However, it will be nice to have the commands available as a KeyboardMarkup sent from the bot so a user can just click on to activate.

This will also feature the help section giving concrete description for each of the available commands.

### PR DESCRIPTION
**Pull request by andrew21-mch**
fixes #31 

Changes coming into this repo from this branch include:
 - Updated the bot.py with a function for keyboard menu

### CODE REVIEW

**Comment by andrew21-mch** [comment]
job
heres is the look

**Comment by andrew21-mch** [comment]
hi @elroykanye this is what i have been able to do so far, i think we can build from here

**Comment by elroykanye** [comment]
Thanks, I will connect to this end this morning

**Comment by andrew21-mch** [comment]
> Thanks, I will connect to this end this morning

ok then good night

**Review by andrew21-mch — COMMENTED** [review]
I am good with this 
Will have to pull it 

@elhmn please you can check this out

**Comment by elroykanye** [comment]
it will not run 😅

**Comment by andrew21-mch** [comment]
> it will not run 😅

Oh haha
We keep working then

**Comment by elroykanye** [comment]
It will run now. The is no problem. Apparently, I renamed every `start` in my project and it affected even the libraries I had.
I just reinit the venv

**Comment by elroykanye** [comment]
Here is the most recent selection menu

Screenshot_20220415-133530

**Comment by elroykanye** [comment]
We'll have to remove this from draft before calling @elhmn to review. 
Let's resolve something in the discussions.

**Comment by andrew21-mch** [comment]
Ok

**Comment by andrew21-mch** [comment]
Guess i did something incorrect, still worked on the help section and pushed to this branch instead, anyway to fix that @elroykanye

**Comment by elroykanye** [comment]
It is fine. That is part of this from the discussion we had.

**Comment by Mankavelda** [comment]
looks nice

File: src/app/bot.py [hunk 1]
```diff
@@ -1,99 +1,129 @@
...
 
 
+def oss_bot_menu(update: Update, context: CallbackContext) -> None:
+    reply_markup = main_menu_keyboard
+    update.message.reply_text('Select Menu', reply_markup=reply_markup)
+
+
 # function to handle the /help command
-def help_cmd(update: Update, context: CallbackContext):
+def oss_bot_help(update: Update, context: CallbackContext):
```
Comment (elhmn) [inline src/app/bot.py:62]: I find the `help_command` name for this function more appropriate. When naming a function it should be easier for the maintainers (someone reading the code) to figure out what it does simply by reading the name of the function.

File: src/app/bot.py [hunk 2]
```diff
@@ -1,99 +1,129 @@
...
     test_poll.pause()
 
 
 # resume game
-def resume(update, context):
+def oss_bot_resume(update: Update, context: CallbackContext) -> None:
     test_poll.resume()
 
 
+def oss_bot_menu(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:56]:
```suggestion
def resume_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py (hunk 1 above)
Comment (elhmn) [inline src/app/bot.py:62]:
```suggestion
def help_command(update: Update, context: CallbackContext):
```

File: src/app/bot.py [hunk 3]
```diff
@@ -1,99 +1,129 @@
...
 
 # pause game
-def pause(update, context):
+def oss_bot_pause(update: Update, context: CallbackContext) -> None:
     test_poll.pause()
 
 
 # resume game
-def resume(update, context):
+def oss_bot_resume(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:52]:
```suggestion
def resume_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 4]
```diff
@@ -1,99 +1,129 @@
...
 
 # schedule a game
-def schedule(update, context):
+def oss_bot_schedule(update: Update, context: CallbackContext) -> None:
     update.message.reply_text("Scheduling games")
 
 
 # pause game
-def pause(update, context):
+def oss_bot_pause(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:47]:
```suggestion
def pause_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 5]
```diff
@@ -1,99 +1,129 @@
...
 # stop ongoing game
-def stop(update, context):
+def oss_bot_stop(update: Update, context: CallbackContext) -> None:
     update.message.reply_text("Stopping game")
     test_poll.stop()
 
 
 # schedule a game
-def schedule(update, context):
+def oss_bot_schedule(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:42]:
```suggestion
def schedule_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 6]
```diff
@@ -1,99 +1,129 @@
...
 
 # Display leaderboard
-def leaderboard(update, context):
+def oss_bot_leaderboard(update: Update, context: CallbackContext) -> None:
     update.message.reply_text("Leaderboard is not implemented yet")
 
 
 # stop ongoing game
-def stop(update, context):
+def oss_bot_stop(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:36]:
```suggestion
def stop_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 7]
```diff
@@ -1,99 +1,129 @@
...
 
 # show available games
-def games(update, context):
+def oss_bot_games(update: Update, context: CallbackContext):
     update.message.reply_text("Games are not implemented yet")
 
 
 # Display leaderboard
-def leaderboard(update, context):
+def oss_bot_leaderboard(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:31]:
```suggestion
def leaderboard_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 8]
```diff
@@ -1,99 +1,129 @@
...
 
-def about(update: Update, context: CallbackContext):
-    update.message.reply_text(f'This is a bot for the {platform.system()} platform')
+def oss_bot_about(update: Update, context: CallbackContext) -> None:
+    update.message.reply_text(f'BOT NAME: {BOT_NAME} \n BOT VERSION: 1.0 \n BOT PLATFORM: {platform.system()} is a quizbot designed to make the community interactive 😂 have fun while developing your skills')
 
 
 # show available games
-def games(update, context):
+def oss_bot_games(update: Update, context: CallbackContext):
```
Comment (elhmn) [inline src/app/bot.py:26]:
```suggestion
def list_games_command(update: Update, context: CallbackContext):
```

File: src/app/bot.py [hunk 9]
```diff
@@ -1,99 +1,129 @@
...
+
+
+def oss_bot_start(update: Update, context: CallbackContext) -> None:
+    logging.info('Starting game')
     update.message.reply_text("Hi\nTo start a game, select game type.")
 
 
-def about(update: Update, context: CallbackContext):
-    update.message.reply_text(f'This is a bot for the {platform.system()} platform')
+def oss_bot_about(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:21]:
```suggestion
def about_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 10]
```diff
@@ -1,99 +1,129 @@
...
 # function to greet the user
 from data.quiz import test_poll
 
-
 # function to handle the /start command
-def start(update: Update, context: CallbackContext):
+from menus import main_menu_keyboard
+
+
+def oss_bot_start(update: Update, context: CallbackContext) -> None:
```
Comment (elhmn) [inline src/app/bot.py:16]:
```suggestion
def start_command(update: Update, context: CallbackContext) -> None:
```

File: src/app/bot.py [hunk 11]
```diff
@@ -1,99 +1,129 @@
...
     update.message.reply_text("Hi\nTo start a game, select game type.")
 
 
-def about(update: Update, context: CallbackContext):
-    update.message.reply_text(f'This is a bot for the {platform.system()} platform')
+def oss_bot_about(update: Update, context: CallbackContext) -> None:
+    update.message.reply_text(f'BOT NAME: {BOT_NAME} \n BOT VERSION: 1.0 \n BOT PLATFORM: {platform.system()} is a quizbot designed to make the community interactive 😂 have fun while developing your skills')
 
 
 # show available games
```
Comment (elhmn) [inline src/app/bot.py:25]: I will recommend that you use docstrings to comment your functions.

You can have a look on this [link](https://www.programiz.com/python-programming/docstrings#:~:text=As%20mentioned%20above%2C%20Python%20docstrings,the%20__doc__%20attribute.) or search more information on the web on how to correctly comment python code

**Review by elhmn — CHANGES_REQUESTED** [review]
Nice work, I left comments regarding how you named your functions and documented the code in general

File: src/app/bot.py (hunk 11 above)
Comment (andrew21-mch) [reply src/app/bot.py:25]: Alright Boris
we will work on that

Comment (andrew21-mch) [reply src/app/bot.py:25]: @Mankavelda Will you like to work on this ?

File: src/app/bot.py (hunk 1 above)
Comment (elroykanye) [reply src/app/bot.py:62]: Okay @elhmn , will do just this

File: src/app/bot.py (hunk 11 above)
Comment (elroykanye) [reply src/app/bot.py:25]: I think @andrew21-mch and I should work on this while @Mankavelda adjusts the changes requested for the function names.

Comment (andrew21-mch) [reply src/app/bot.py:25]: ok @elroykanye

Comment (elhmn) [reply src/app/bot.py:25]: Wait are you all going to work on the same pull request ?

Don't you think it is faster if one person (the author of the PR) sort it out himself ?

Comment (andrew21-mch) [reply src/app/bot.py:25]: Its cool, let me take on that since i started

Comment (elroykanye) [reply src/app/bot.py:25]: We are actually trying to get @Mankavelda involved and participate more 😅 I actually feel off about more than 2 people on  the same pr but I think we will go another way

**Comment by andrew21-mch** [comment]
Hi @elhmn I just did some changes now, is this okay now ?

File: src/app/bot.py [hunk 12]
```diff
@@ -1,99 +1,135 @@
 import logging
+import platform
+
+from django import dispatch
 
-from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext
+from telegram import InlineKeyboardButton, InlineKeyboardMarkup, ReplyKeyboardMarkup, KeyboardButton
+from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext, CallbackQueryHandler
 from telegram.update import Update
```
Comment (elhmn) [inline src/app/bot.py:8]: is these dependencies `[django, and telegram]` in the `requirements.txt` file ? You can run `python -m pip freeze > requirements.txt` to be sure

**Review by elhmn — APPROVED** [review]
It looks way better, I left a comment about keeping the `requirements.txt` up to date.

File: src/app/bot.py (hunk 11 above)
Comment (elhmn) [reply src/app/bot.py]: The best way to have @Mankavelda involved is to assign her to a separate task.

File: src/app/bot.py (hunk 12 above)
Comment (elroykanye) [reply src/app/bot.py]: Updated master and merged into this to bring the requirements generated from the other PR. 
`django` was a mistake I think. :-|

**Review by elroykanye — APPROVED** [review]

**Pull request merged**